adhocteam/Head-Start-TTADP
# Init CD

**Description of change**
In short, this code adds an automated deployment pipeline.
- Add deploy job to circle ci config
- Call deploy job only on deployment branches (adhoc fork `main` for dev, HHS `main` for staging, HHS `prod` for prod)
- Change frontend `yarn build` command to change build output dir, tell `index.js` about this new dir location
- Add backend `yarn build` command to create compiled copy of BE files, changes `yarn start` to use new compiled index.js

**NOTE: Before this code is merged to the upstream HHS/Head-Start-TTADP repo, the HHS project circle ci env variables must be added for govcloud's deployer username/password and space name.** 

**How to test**
Check out latest build for this branch on circle ci https://app.circleci.com/pipelines/github/adhocteam/Head-Start-TTADP?branch=sj-cd-init
This works because this branch, `sj-cd-init`, is set to deploy to cloud.gov's sandbox environment. Before this PR is merged all sandbox specific deployment code will be removed.

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/12

**Checklist**

- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
